coordinator/gscoordinator/coordinator.py

```diff
@@ -760,6 +791,35 @@ def FetchLogs(self, request, context):
                         info_message=info_message, error_message=error_message
                     )
 
+    def AddLib(self, request, context):
+        if request.session_id != self._session_id:
+            context.set_code(grpc.StatusCode.INVALID_ARGUMENT)
+            context.set_details(
+                f"Session handle not matched, {request.session_id} versus {self._session_id}"
+            )
+        os.makedirs(self._resource_dir, exist_ok=True)
+        gar = request.gar
+        fp = BytesIO(gar)
+        filename = None
+        with zipfile.ZipFile(fp, "r") as zip_ref:
+            zip_ref.extractall(self._resource_dir)
+            logger.info(
+                "Coordinator recieved add lib request contains file {}".format(
+                    zip_ref.namelist()
+                )
+            )
+            if len(zip_ref.namelist()) != 1:
+                raise RuntimeError("Expect only one resource in one gar")
+            filename = zip_ref.namelist()[0]
+        full_filename = os.path.join(self._resource_dir, filename)
+        self._launcher.distribute_file(full_filename)
+        logger.info("Successfully distributed {}".format(full_filename))
+        if full_filename.endswith(".jar"):
+            logger.info("adding lib to java class path since it ends with .jar")
+            self._java_class_path = full_filename + ":" + self._java_class_path
```

Comment:
Can we fixes the classPath and no need to change it every time during `AddLib`

Reply:
In our design, User can upload/add jars via python client interactively, so classPath need to be upated.